My UltraSPARC U5 panics on boot about a memory alignment error when compiled
with gcc4 and -O. Compiling with -O2 fixes this, and it should not adversly
affect debugging.

NOTE: This does not fix the memory alignment panic when loading kernel modules.

Roy Marples (uberlord@gentoo.org)

diff -ur a/conf/kern.pre.mk b/conf/kern.pre.mk
--- a/conf/kern.pre.mk	2007-05-06 21:36:18 +0100
+++ b/conf/kern.pre.mk	2007-05-09 09:58:37 +0100
@@ -19,15 +19,10 @@
 .if ${CC} == "icc"
 COPTFLAGS?=	-O
 .else
-. if defined(DEBUG)
-_MINUS_O=	-O
-. else
-_MINUS_O=	-O2
-. endif
 . if ${MACHINE_ARCH} == "amd64"
 COPTFLAGS?=-O2 -frename-registers -pipe
 . else
-COPTFLAGS?=${_MINUS_O} -pipe
+COPTFLAGS?=-O2 -pipe
 . endif
 . if !empty(COPTFLAGS:M-O[23s]) && empty(COPTFLAGS:M-fno-strict-aliasing)
 COPTFLAGS+= -fno-strict-aliasing
